packages/medusa/src/api-v2/store/payment-collections/validators.ts

```diff
@@ -0,0 +1,30 @@
+import { z } from "zod"
+import { createSelectParams } from "../../utils/validators"
+
+export type StoreGetPaymentCollectionParamsType = z.infer<
+  typeof StoreGetPaymentCollectionParams
+>
+export const StoreGetPaymentCollectionParams = createSelectParams()
+
+export type StoreCreatePaymentSessionType = z.infer<
+  typeof StoreCreatePaymentSession
+>
+export const StoreCreatePaymentSession = z
+  .object({
+    provider_id: z.string(),
+    context: z.record(z.unknown()).optional(),
+    data: z.record(z.unknown()).optional(),
+  })
+  .strict()
+
+export type StoreCreatePaymentCollectionType = z.infer<
+  typeof StoreCreatePaymentCollection
+>
+export const StoreCreatePaymentCollection = z
+  .object({
+    cart_id: z.string(),
+    region_id: z.string(),
+    currency_code: z.string(),
+    amount: z.number(),
```

Comment:
thought: Think we should remove `amount` - you shouldn't be able to pass your own amount from the storefront.

Reply:
Good point, I just took how it was before, but I completely agree 👍